File: grappling_hook/csharp prototype/grapple_ui/models/viewmodels/GrappleArrows.cs
namespace models.viewmodels
{
    /// <summary>
    /// This is the arrow graphic drawn next to the stick figure graphic
    /// </summary>
    /// <remarks>
    /// NOTE: The positions are offsets from center
    /// </remarks>
    public class GrappleArrows
    {
        /// <summary>
        /// Whether to show the look arrow (should only show it when they have acceleration along look)
        /// </summary>
        public bool showLook { get; set; }

        // True: Use the standard color
        // False: Use the gray color
        public bool isStandardColor_primary { get; set; }
        public bool isStandardColor_look { get; set; }

        // Set to true when they hover over options that will modify these lines
        public bool isHighlight_primary { get; set; }
        public bool isHighlight_look { get; set; }

        // Positions
        public double primary_from_x { get; set; }
        public double primary_to_x { get; set; }

        public double primary_y { get; set; }      // if the primary was allowed a slope, it would complicate lots of other drawing (they would need to calculate perpendicular lines, parallel lines)

        public double look_from_x { get; set; }
        public double look_from_y { get; set; }

        public double look_to_x { get; set; }
        public double look_to_y { get; set; }
    }
}

File: grappling_hook/cet/ui_controls_grapple/grapple_arrows.lua
-- def is models\viewmodels\GrappleArrows
-- style_graphics is models\stylesheet\Graphics
function Draw_GrappleArrows(def, style_graphics, screenOffset_x, screenOffset_y, parent_width, parent_height, scale)
    local center_x = parent_width / 2
    local center_y = parent_height / 2

    -- Primary
    local color
    if def.isHighlight_primary then
        color = style_graphics.line_color_highlight_abgr
    elseif def.isStandardColor_primary then
        color = style_graphics.line_color_standard_abgr
    else
        color = style_graphics.line_color_gray_abgr
    end

    Draw_Arrow(screenOffset_x, screenOffset_y, center_x + (def.primary_from_x * scale), center_y + (def.primary_y * scale), center_x + (def.primary_to_x * scale), center_y + (def.primary_y * scale), color, style_graphics.line_thickness_main, style_graphics.arrow_length * scale, style_graphics.arrow_width * scale)

    -- Look
    if def.showLook then
        if def.isHighlight_look then
            color = style_graphics.line_color_highlight_abgr
        elseif def.isStandardColor_look then
            color = style_graphics.line_color_standard_abgr
        else
            color = style_graphics.line_color_gray_abgr
        end

        Draw_Arrow(screenOffset_x, screenOffset_y, center_x + (def.look_from_x * scale), center_y + (def.look_from_y * scale), center_x + (def.look_to_x * scale), center_y + (def.look_to_y * scale), color, style_graphics.line_thickness_main, style_graphics.arrow_length * scale, style_graphics.arrow_width * scale)
    end
end